packages/wallets/src/features/cashier/flows/WalletWithdrawal/WalletWithdrawal.tsx

```diff
@@ -33,23 +38,33 @@ const WalletWithdrawal = () => {
 
     const isCrypto = activeWallet?.currency_config?.is_crypto;
 
+    const onCloseHandler = () => setVerificationCode('');
+
+    const resetError = () => {
+        onCloseHandler();
+        setError(undefined);
+    };
+
+    if (isServerError(error)) {
+        return <WithdrawalErrorScreen error={error} resetError={resetError} setResendEmail={setResendEmail} />;
+    }
+
     if (verificationCode) {
         if (isCurrencyConfigSuccess && activeWallet?.currency) {
             return isCrypto ? (
                 <WithdrawalCryptoModule
-                    onClose={() => {
-                        setVerificationCode('');
-                    }}
+                    onClose={onCloseHandler}
+                    setError={setError}
                     verificationCode={verificationCode}
                 />
             ) : (
-                <WithdrawalFiatModule verificationCode={verificationCode} />
+                <WithdrawalFiatModule setError={setError} verificationCode={verificationCode} />
```

Comment:
would it be better to move error state ownership to WalletWithdrawal, instead of passing an callback to WithdrawalFiatModule just for it to load data and pass the result back WalletWithdrawal? Just load data and handle errors within WalletWithdrawal itself (no passing back and forth, and especially no need to pass data down which is gateway drug towards general data handling problems)

Reply:
yes @wojciech-deriv it make sense to do this, even i thought about it
but here is one concern. again i believe we had the same discussion and concerns
since we want to make WithdrawalFiat and WithdrawalCrypto as a modules we need to store API calls inside the modules and set the errors there.
One way that i can imagine how to avoid this is to keep error state handling in both: WithdrawalFiat and WithdrawalCrypto components
but
1. we will end up with duplicated code
2. we still need to pass the props such as setResendEmail because EmailVerification is in the withdrawal flow scope.
again this discussion is going into what we should provide to reuse? Module or a Flow? Both? etc.

Reply:
I think passing "setResentEmail" is fine - thats a callback on user action, thats what callbacks are for. 
Passing "setError" is not fine - thats a function to transfer data from server response up the component hierarchy. Props are meant for data transfer beetween components, state management libraries are meant to do so.

either way, I get that it would require major refactors to change that, so probably out of scope :-/